### DISCUSSION TITLE
Autocomplete `select`s that send ID values rather than strings

### DISCUSSION THREAD

**Discussion by nimmolo** [MushroomObserver/mushroom-observer#1433] [Ideas]
Here's an idea i had to lighten server load for the "Observations needing Identification" filters, but it may be applicable elsewhere.

Currently, our autocomplete fields send AJAX requests, and the controller handling them tries to match the user's string against a db column (like `observations.text_name`), then the controller returns an array of matching strings from the column. 
```js
[
"Amanita",
"Amanitaceae",
"Amanitoides"
]
```
If the user accepts a string match and submits the form, the form will send that string as a param, and the controller response has to look up the string again to try to match it to a model instance.

What if the form could send the model ID instead of the string, as the `param`? 

I'm thinking to write a second type of AJAX response that instead of an array of strings, returns a JSON object, keyed by the db ID, instead. That way, the UX could be a "autocompleted `select`" that sends the ID as the value selected, avoiding the second lookup. The user's choices would be truly constrained by available options, which in the case of searches is sometimes what we want.

The AJAX response would send an object like this to populate the `select` options in the dropdown:
```js
[
{ value: "1234", text:  "Amanita" },
{ value: "234", text:  "Amanitaceae" },
{ value: "3467", text: "Amanitoides"}
]
```

Here's the [UI pattern](https://bootstrap-autocomplete.readthedocs.io/en/latest/), that i imagine we've all used elsewhere on the web.

**Comment by mo-nathan** [comment]
Sounds worth pursuing.  I think I see where you're going and I think it make sense.
